### PR TITLE
Bug 2037274: starter.go: add invalidCertsController

### PR DESCRIPTION
**Pull request by s-urbaniak**
This includes a controller detecting invalid provider certificates as described in https://github.com/openshift/enhancements/pull/980.

### CODE REVIEW

**Comment by s-urbaniak** [comment]
1.  If the Bugzilla associated with the PR has the "FastFix" keyword, the subjective assessment on the issue has already been done and a customer is impacted. These PRs should be prioritized for merge.
    - [x] verified
    - [ ] does not apply
2. The bug has significant impact either through severity, reduction in supportability, or number of users affected.
    - [x] verified
    - [ ] does not apply
3. For branches that are in the Maintenance lifecycle phase:
    - [x] The bug is a critical fix, no reasonable workaround exists, and a recommendation for upgrade has been ruled out, or 
    - [ ] The bug is a security related bug
    - [ ] Branch not in maintenance mode yet (current release + previous release for 90 days after current GA; everything older is in maintenance)
4. The severity field of the bug must be set to accurately reflect criticality.
    - [x] verified
5. The PR was created with the cherry-pick bot OR the PR’s description is well formed with user-focused release notes that state the bug number, impact, cause, and resolution. Where appropriate, it should also contain information about how a user can identify whether a particular cluster is affected.
    - [x] verified

File: pkg/operator/starter.go [hunk 1]
```diff
@@ -223,6 +224,17 @@ func prepareOauthOperator(controllerContext *controllercmd.ControllerContext, op
 
 	oauthInformers := oauthinformers.NewSharedInformerFactory(oauthClient, resync)
 
+	invalidCertsController := metricscontroller.NewMetricsController(
+		"InvalidProviderCertsController",
```
Comment (stlaz) [inline pkg/operator/starter.go:228]: Given that the only condition message we'll get is: `<controllerName>::InvalidCertsDetected: x server certificates without SAN detected`, should we maybe call this something else?

Comment (s-urbaniak) [reply pkg/operator/starter.go:228]: let's call it `InvalidCertsController`, similar to cluster-kube-apiserver-operator we might have more metrics in here.

Comment (s-urbaniak) [reply pkg/operator/starter.go:228]: the discussion results are manifested in https://github.com/openshift/library-go/pull/1289

**Pull request merged**